arch/arm/core/aarch32/cortex_a_r/vector_table.S

```diff
@@ -23,5 +23,9 @@ SECTION_SUBSEC_FUNC(exc_vector_table,_vector_table_section,_vector_table)
 	ldr pc, =z_arm_prefetch_abort    /* prefetch abort offset  0xc */
 	ldr pc, =z_arm_data_abort        /* data abort     offset 0x10 */
 	nop				 /*                offset 0x14 */
+#ifdef CONFIG_GEN_SW_ISR_TABLE
 	ldr pc, =_isr_wrapper		 /* IRQ            offset 0x18 */
+#else
+	ldr pc, =z_irq_spurious
```

Comment:
should this be handled in Kconfig with a depends instead ?
https://github.com/zephyrproject-rtos/zephyr/blob/724601bfea5dfb6fd84159efb928f6e37273d4d9/arch/Kconfig#L333-L336

seems wrong to me that users can disable `GEN_SW_ISR_TABLE` if that means you will end in `z_irq_spurious`.

I think that will be hard for a user to find debug.

Reply:
I don't know, but the problem happens even when you disable `GEN_ISR_TABLES`. The assembler is unconditionally looking that symbol that exists only when you enable it. `z_irq_spurious` will cause a fault that seems better than just a silent misbehavior.

Reply:
> z_irq_spurious will cause a fault that seems better than just a silent misbehavior.

Agreed, but to me it seems even better if we can avoid getting into that situation in the first place, by having appropriate Kconfig dependencies.
If that is possible ?

Reply:
we have Kconfig dependencies handling it, I don't know what you are proposing The problem is that the code is not checking this dependency. As far as I  understand there is nothing we can do, if an interruption happens and we don't have a handler it will cause a failure, this code is just adding a default handler for interruptions that is `z_irq_spurious` in case we build without generate an interruption table. Am I missing something ?

Reply:
> Am I missing something ?

No, just took a look at the whole file.
This is the right thing to do.